packages/core/src/lib/wallet-selector.ts

```diff
@@ -48,6 +48,12 @@ export const setupWalletSelector = async (
 
         return wallet;
       },
+      setActiveAccount: (accountId: string) => {
+        store.getState().accounts = store.getState().accounts.map((account) => {
+          account.active = account.accountId === accountId;
+          return account;
+        });
```

Comment:
To update the state, we must dispatch a new action so it can go through the store reducer. It's important that this happens instead of mutating like this as it allows subscribes to be notified